src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -9288,6 +9288,20 @@ static void (*D_80854738[])(PlayState* play, Player* this) = {
 
 static Vec3f D_80854778 = { 0.0f, 50.0f, 0.0f };
 
+u32 Player_GetLargestGI(void) {
+    u32 maxObjSize = 0, curObjSize = 0;
+    for (s32 i = 0; true; i++) {
```

Comment:
I'm not a huge fan of infinite loops tbh, well it's not an infinite loop since there's a return but idk, I'd prefer something like this:
```suggestion
    for (s32 i = OBJECT_ID_MAX; i == OBJECT_INVALID; i--) {
```
the idea is starting from the the max ID and going in reverse through the object list, if I'm understanding this correctly the loop goes through the objects to get the max size value

Reply:
well, it should be ``s32 i = OBJECT_ID_MAX - 1`` actually but you get the idea

Reply:
I think you’re misunderstanding. The loop iterator is not the object ID, that is being read from the table, so this code would not work as is.

Reply:
oh yea you're right lol, so there's no alternative to this pseudo infinite loop?

Reply:
Not really, you could loop as many times as there are entries in the table, but the table ends when there is OBJECT_ID_INVALID, which is reflected in the code here. I personally feel that this is the most elegant way to do it, hence why I did it this way, but open to suggestions.